### DISCUSSION TITLE
how to set up init_batch_size

### DISCUSSION THREAD

**Discussion by imindshejian** [deepmodeling/dpgen#370] [Q&A] [answered]
i have 4 sets in init_sets, which contains 2000frames of 120 atoms, in the README of dpgen, one recommended rule for setting the sys_batch_size and init_batch_size is that batch_size mutiply number of atoms ot the stucture should be larger than 32. If set to auto, batch size will be 32 divided by number of atoms. For example, if I set init_batch_size as 4, does it mean i put 480 atoms train togerther, and i found that no matter how init_batch_size is the GPU memory were all ouccpied during training, but the Volatile GPU-util increased.
 What is the relationship between init_batch_size for init_set and batch_size in training section? Should i set a larger init_batch_size accoording to the Volatile GPU-util? Does it mean a larger init_batch_size increase training efficiency?

**Comment by amcadmus** [comment, marked as answer]
The “auto” style finds the minimal batch size so that the batch size times number of atoms in a frame is larger than 32. 

In your case, you have 120 atoms in each frame, the batch size will be set to 1, because 1 x 120 >32.
